File: api/step.py
from core.turing_machine import TuringMachine
import json
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))


def handler(event, context):
    """Executa um passo da máquina"""
    try:
        if isinstance(event.get('body'), str):
            body = json.loads(event['body'])
        else:
            body = event.get('body', {})

        machine_data = body.get('machine')

        if not machine_data:
            return {
                'statusCode': 400,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps({
                    'success': False,
                    'error': 'Dados da máquina ausentes'
                })
            }

        tm = TuringMachine.from_dict(machine_data)
        tm.step()

        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'success': True,
                'machine': tm.to_dict()
            })
        }

    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'success': False,
                'error': f'Erro no servidor: {str(e)}'
            })
        }

File: api/step.js
import { TuringMachine } from '../core/turing_machine.js';

export default function handler(req, res) {
  try {
    const { machine } = req.body;

    if (!machine) {
      return res.status(400).json({
        success: false,
        error: 'Dados da máquina ausentes'
      });
    }

    const tm = TuringMachine.fromDict(machine);
    tm.step();

    return res.status(200).json({
      success: true,
      machine: tm.toDict()
    });
  } catch (e) {
    return res.status(500).json({
      success: false,
      error: `Erro no servidor: ${e.message}`
    });
  }
}
